### PR TITLE
[FLINK-20150][debezium-avro] Add documentation for the debezium-avro-confluent format

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

Add documentation for the `debezium-avro-confluent` format.

## Brief change log

- Update the existing `debezium.md` page. 

## Verifying this change

N/A

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
Hi @danny0405 , could you help to review this doc?

File: docs/dev/table/connectors/formats/debezium.md [hunk 1]
```diff
@@ -154,6 +162,50 @@ SELECT * FROM topic_products;
 Format Options
 ----------------
 
+Flink provides `debezium-avro-confluent` and `debezium-json` formats to interpret Avro or Json messages produced by Debezium.
+Please pick corresponding format according to your Debezium message encoding.
+
```
Comment (danny0405) [inline docs/dev/table/connectors/formats/debezium.md:167]: Use format `debezium-avro-confluent` to interpret Debezium Avro messages and format `debezium-json` to interpret Debezium Json messages.

File: docs/dev/table/connectors/formats/debezium.md [hunk 2]
```diff
@@ -99,7 +107,7 @@ CREATE TABLE topic_products (
  'topic' = 'products_binlog',
  'properties.bootstrap.servers' = 'localhost:9092',
  'properties.group.id' = 'testGroup',
- 'format' = 'debezium-json'  -- using debezium-json as the format
+ 'format' = 'debezium-json'  -- please use 'debezium-avro-confluent' if Debezium encodes messages in Avro format
 )
```
Comment (danny0405) [inline docs/dev/table/connectors/formats/debezium.md:111]: We better keep the original comment and add a new TAB for avro here.

**Comment by wuchong** [comment]
@danny0405 , hope I have addressed your concerns.

**Review by danny0405 — APPROVED** [review]
+1, looks good overall.

**Comment by wuchong** [comment]
Thanks @danny0405 .

**Pull request merged**